packages/plugin-conversation/src/conversation.js

```diff
@@ -0,0 +1,607 @@
+            if (userId) {
+              Object.assign(params, {
+                api: `conversation`,
+                resource: `conversations/user/${userId}`
+              });
+            }
+            else {
+              params.uri = url;
+            }
+            return this.request(params);
+          });
+      })
+      .then((res) => this._recordUUIDs(res.body)
+      .then(() => res.body));
+  },
+
+  leave(conversation, object, activity) {
+    this._inferConversationUrl(conversation);
+    if (!object) {
+      object = this.spark.device.userId;
+    }
+
+    return this.spark.user.asUUID(object)
+      .then((id) => this.prepare(activity, {
+        verb: `leave`,
+        target: this.prepareConversation(conversation),
+        object: {
+          id,
+          objectType: `person`
+        },
+        kmsMessage: {
+          method: `delete`,
+          uri: `<KRO>/authorizations?${querystring.stringify({authId: id})}`
+        }
+      }))
+      .then((a) => this.submit(a));
+  },
+
+  list(options) {
+    return this._list({
+      api: `conversation`,
+      resource: `conversations`,
+      qs: options
+    });
+  },
+
+  listLeft(options) {
+    return this._list({
+      api: `conversation`,
+      resource: `conversations/left`,
+      qs: options
+    });
+  },
+
+  listActivities(options) {
+    return this._listActivities(Object.assign(options, {mentions: false}));
+  },
+
+  listMentions(options) {
+    return this._listActivities(Object.assign(options, {mentions: true}));
+  },
+
+  /**
+   * Mutes the mentions of a conversation
+   * @param {Conversation~ConversationObject} conversation
+   * @param {Conversation~ActivityObject} activity
+   * @param {Object} options
+   * @returns {Promise} Resolves with the created activity
+   */
+  muteMentions(conversation, activity, options) {
+    return this.tag(conversation, {
+      tags: [`MENTION_NOTIFICATIONS_OFF`]
+    }, activity, options);
+  },
+
+  /**
+   * Mutes the messages of a conversation
+   * @param {Conversation~ConversationObject} conversation
+   * @param {Conversation~ActivityObject} activity
+   * @param {Object} options
+   * @returns {Promise} Resolves with the created activity
+   */
+  muteMessages(conversation, activity, options) {
+    return this.tag(conversation, {
+      tags: [`MESSAGE_NOTIFICATIONS_OFF`]
+    }, activity, options);
+  },
+
+  post(conversation, object, activity) {
+    return this._inferConversationUrl(conversation)
+      .then(() => this.prepare(activity, {
+        verb: `post`,
+        target: this.prepareConversation(conversation),
+        object: defaults(object, {objectType: `comment`})
+      }))
+      .then((a) => this.submit(a));
+  },
+
+  prepareConversation(conversation) {
+    const c = defaults(pick(conversation, `id`, `url`, `objectType`, `kmsResourceObjectUrl`, `defaultActivityEncryptionKeyUrl`), {
+      objectType: `conversation`
+    });
+    return c;
+  },
+
+  prepare(activity, params) {
```

Comment:
good point; i've made a note to handle it in one of the later PRs (it'll really mess up the rebase for the already pending PRs)